jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -81,8 +79,13 @@ object XGBoost extends Serializable {
       trainingSet: RDD[MLLabeledPoint],
       params: Map[String, Any],
       rabitEnv: java.util.Map[String, String],
-      numWorkers: Int, round: Int, obj: ObjectiveTrait, eval: EvalTrait,
-      useExternalMemory: Boolean, missing: Float = Float.NaN): RDD[Booster] = {
+      numWorkers: Int,
+      round: Int,
+      obj: ObjectiveTrait,
+      eval: EvalTrait,
+      useExternalMemory: Boolean,
+      missing: Float,
+      baseMargin: RDD[Float]): RDD[Booster] = {
```

Comment:
I've seen the discussion in #2223 and decided not to modify LabeledPoint at the moment. This could be done in a separate PR for both `baseMargin` and `groupData`.